### ISSUE TITLE
Plugin Quick Picks Never Update `.value`

### ISSUE THREAD

**Issue by colin-grant-work** [eclipse-theia/theia#10186]
### Bug Description:

If you set the `.value` field of a QuickPick from a plugin, that value is never updated, regardless of what the user enters. In VSCode, the field tracks the user's input.

### Steps to Reproduce:

1. Fetch [this branch](https://github.com/colin-grant-work/vscode-extension-samples/tree/demo/unchanging-value)
1a. Or apply the changes in [this commit](https://github.com/colin-grant-work/vscode-extension-samples/commit/f2907a884ed90205b67358ffac8ae1c75e40fb1e) to your local fork.
2. Build the `quickinput-sample` plugin with `vsce package`
3. Move the built `.vsix` file into the `plugins` folder of your local Theia
4. Start the application
5. Open the command palette and run the command 'Quick Input Samples'
6. The value in the quick input should be `Initial value`.
7. Enter some other value, for example `Something else`.
8. Hit enter.
9. Observe that you see a toast with the message `You entered: Initial value`.

VSCode:  
![image](https://user-images.githubusercontent.com/62660806/135176176-d5d0037a-36a7-4fcf-a098-2a473fe1049c.png)

Theia:  
![image](https://user-images.githubusercontent.com/62660806/135176311-ee9e1578-3cc7-4aa4-a42b-ed8b88e3a82c.png)

### Additional Information

- Operating System: RHEL7
- Theia Version: 956b93e8c438b35b7ba1bf256a422f59b2e76308

**Comment by tsmaeder** [comment]
Just to clarify: the plugin sets the initial `.value` field, the user types something into the input (or selects?) and then the plugin reads the `.value` field, but the value the plugin reads is the initial value, correct?

**Comment by colin-grant-work** [comment]
@tsmaeder, yes that's exactly right. If the plugin sets an initial `.value` after using the `createQuickPick()` method, then that `.value` field is fixed, regardless of what the user types.
